### PR TITLE
Swagger spec added

### PR DESCRIPTION
**Pull request by ansultan1**
i have add a swagger yaml file and generated the  swagger client and server additionally in the db layer i have add the mongodb client as well.

### CODE REVIEW

File: db/client.go [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+	"log"
+	"task-scheduler/models"
+)
+
+type DataStore interface {
+	AddOrUpdateTask(task *models.Task) (string, error)
+	GetTask(id string) (*models.Task, error)
+	ListTasks() ([]*models.Task, error)
+	DeleteTask(id string) error
+	//UpdateTask(task *models.Task) error
```
Comment (OvaisTariq95) [inline db/client.go]: why commentd out ?

File: db/mongo/client.go [hunk 2]
```diff
@@ -0,0 +1,108 @@
+package mongo
+
+import (
+	"context"
+	"fmt"
```
Comment (OvaisTariq95) [inline db/mongo/client.go:5]: sort imports

File: db/mongo/client_test.go [hunk 3]
```diff
@@ -0,0 +1,164 @@
+package mongo
+
+import (
+	"fmt"
+	"os"
+	"task-scheduler/db"
```
Comment (OvaisTariq95) [inline db/mongo/client_test.go]: sort imports

**Review by OvaisTariq95 — CHANGES_REQUESTED** [review]

File: db/mongo/client.go (hunk 2 above)
Comment (ansultan1) [reply db/mongo/client.go:5]: Previously i used the gofmt utility which basically sort the imports in alphabetical order. Now i used the goimports utility which group the imports like on local, standard and third party packages and then sort them .

Comment (OvaisTariq95) [reply db/mongo/client.go:5]: still wrong the order is as follows 
1. system imports (fmt,context,utils etc)
2. Third party imports("github.com/pkg/errors","github.com/spf13/viper")
3. Your project imports/aliases ("task-scheduler/config","task-scheduler/db")

**Review by OvaisTariq95 — CHANGES_REQUESTED** [review]

File: db/mongo/client.go (hunk 2 above)
Comment (ansultan1) [reply db/mongo/client.go:5]: it was done by the utility automatically. now when i change the order the IDE give me suggestions to sort it.

**Pull request merged**